Build fails on Windows

Running the build script fails on Windows (using MinGW):

```
npm ERR! leveldown@1.4.3 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the leveldown@1.4.3 install script.
npm ERR! This is most likely a problem with the leveldown package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls leveldown
npm ERR! There is likely additional logging output above.
npm ERR! System Windows_NT 6.3.9600
npm ERR! command "c:\\Program Files (x86)\\JXcore\\jx.exe" "c:\\Users\\<username>\
\.jx\\npm" "install" "--build-from-source"
npm ERR! cwd c:\work\projects\thali\ThaliTest\www\jxcore
npm ERR! node -v v0.10.40
npm ERR! npm -v 1.4.14
npm ERR! code ELIFECYCLE
```

Related: #394 

<!---
@huboard:{"order":4.656612873077393e-10,"milestone_order":422,"custom_state":""}
-->

```
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe". To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visual Studio 2005 or 3) add the location of the compon
ent to the system path if it is installed elsewhere.  [C:\Users\vjrantal\AppData\Local\Temp\leveldown-mobile-test\node_modules\leveldown-mobile\build\binding.sln]
MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe". To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visual Studio 2005 or 3) add the location of the compon
ent to the system path if it is installed elsewhere.  [C:\Users\vjrantal\AppData\Local\Temp\leveldown-mobile-test\node_modules\leveldown-mobile\build\binding.sln]
```
from @vjrantal 's gist.

--- a/test/www/jxcore/bv_tests/testThaliReplicationManager.js
+++ b/test/www/jxcore/bv_tests/testThaliReplicationManager.js
@@ -21,9 +21,7 @@ var ThaliReplicationManager = require('thali/thalireplicationmanager');
 // will not interfere with any other databases that might be created
 // during other tests.
 var dbPath = path.join(testUtils.tmpDirectory(), 'pouch-for-replication-test');
-var LevelDownPouchDB = process.platform === 'android' || process.platform === 'ios' ?
-    PouchDB.defaults({db: require('leveldown-mobile'), prefix: dbPath}) :
-    PouchDB.defaults({db: require('leveldown'), prefix: dbPath});
+var LevelDownPouchDB = PouchDB.defaults({db: require('leveldown-mobile'), prefix: dbPath});
 
 // A variable accessible from the tests and from the setup/teardown functions
 // that can be used to make sure that replication managers created during

--- a/test/www/jxcore/package.json
+++ b/test/www/jxcore/package.json
@@ -25,7 +25,6 @@
     "express-pouchdb": "^0.14.2",
     "fs-extra-promise": "^0.2.0",
     "is-property": "^1.0.2",
-    "leveldown": "^1.3.0",
     "leveldown-mobile": "^1.0.7",
     "lie": "^3.0.1",
     "minimist": "^1.2.0",

--- a/thali/install/setUpDesktop.sh
+++ b/thali/install/setUpDesktop.sh
@@ -14,4 +14,4 @@ cd install
 jx npm install
 cd ../../test/www/jxcore
 jx npm link thali
-jx npm install
+jx npm install --no-optional

--- a/thali/install/setUpTests.sh
+++ b/thali/install/setUpTests.sh
@@ -26,6 +26,10 @@ cordova create ThaliTest com.test.thalitest ThaliTest
 mkdir -p ThaliTest/thaliDontCheckIn/localdev
 
 if [ $runningInMinGw == true ]; then
+    # The thali package might be installed as link and there will
+    # be troubles later on if this link is tried to be copied so
+    # remove it here.
+    rm -rf $repositoryRoot/test/www/jxcore/node_modules/thali
     cp -R $repositoryRoot/test/www/ ThaliTest/
 else
     cp -R $repositoryRoot/test/www/ ThaliTest/www
@@ -49,19 +53,14 @@ fi
 # SuperTest which is used by some of the BVTs include a PEM file (for private
 # keys) that makes Android unhappy so we remove it below in addition to the gz
 # files.
-jx npm install --autoremove "*.gz,*.pem"
+jx npm install --no-optional --autoremove "*.gz,*.pem"
 
 # In case autoremove fails to delete the files, delete them explicitly.
-# However, 'find' command on MinGW is likely to consider the predicate '-delete'
-# invalid. In addition, in Windows file system the ThaliTest paths are usually
-# too long and if rm fails, it will abort the script. Thus, on Windows we have
-# to rely on autoremove to work. 
-if [ $runningInMinGw == false ]; then
-    find . -name "*.gz" -delete
-    find . -name "*.pem" -delete
-fi
+find . -name "*.gz" -delete
+find . -name "*.pem" -delete
 
 cp -v $1 app.js
+
 cordova build android --release --device
 
 if [ $runningInMinGw == false ]; then

